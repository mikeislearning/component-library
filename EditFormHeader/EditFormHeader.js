import React from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import { PropTypes } from 'prop-types';
import { Loading } from 'app/components';
import { images } from 'app/theme';

import styles from './EditFormHeader.styles';

export default class EditFormHeader extends React.Component {
  static propTypes = {
    isLoading: PropTypes.bool,
    isValidForm: PropTypes.bool,
    onClose: PropTypes.func,
    onSave: PropTypes.func,
    title: PropTypes.string,
  };

  static defaultProps = {
    isLoading: false,
  };

  renderRightAction() {
    const { isLoading, isValidForm, onSave } = this.props;
    const rightActionTextStyles = !isValidForm ? styles.rightActionDisabled : {};
    const rightAction = <Text style={[styles.rightActionText, rightActionTextStyles]}>Save</Text>;

    if (isLoading) {
      return (
        <View style={[styles.rightActionContainer, styles.loadingContainer]}>
          <Loading size="small" />
        </View>
      );
    } else {
      if (isValidForm) {
        return (
          <TouchableOpacity onPress={onSave} style={styles.rightActionContainer}>
            {rightAction}
          </TouchableOpacity>
        );
      } else if (onSave) {
        return <View style={styles.rightActionContainer}>{rightAction}</View>;
      }
    }

    return null;
  }

  render() {
    const { onClose, title } = this.props;

    return (
      <View style={styles.container}>
        <TouchableOpacity
          accessibilityLabel="Close"
          onPress={onClose}
          style={styles.leftActionContainer}
        >
          <Image source={images.close} style={styles.closeIcon} />
        </TouchableOpacity>
        {title && <Text style={styles.title}>{title}</Text>}
        {this.renderRightAction()}
      </View>
    );
  }
}
