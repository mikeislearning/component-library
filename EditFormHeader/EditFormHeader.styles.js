import { StyleSheet } from 'react-native';
import { colors, fonts } from 'app/theme';

const bottomPadding = 12;

const styles = StyleSheet.create({
  closeIcon: {
    height: 17,
    width: 17,
  },
  container: {
    alignItems: 'flex-end',
    backgroundColor: colors.offwhite,
    borderBottomColor: colors.gray.light,
    borderBottomWidth: 1,
    flexDirection: 'row',
    height: 57,
    justifyContent: 'center',
    paddingBottom: bottomPadding,
  },
  leftActionContainer: {
    bottom: bottomPadding,
    left: 0,
    paddingBottom: 5,
    paddingLeft: 20,
    position: 'absolute',
  },
  loadingContainer: {
    bottom: 0,
    padding: 0,
  },
  rightActionContainer: {
    bottom: bottomPadding,
    paddingRight: 20,
    position: 'absolute',
    right: 0,
  },
  rightActionDisabled: {
    color: colors.gray.medium,
  },
  rightActionText: {
    color: colors.blue,
    fontFamily: fonts.family.messinaSans.semiBold,
    fontSize: 18,
    lineHeight: 24,
  },
  title: {
    color: colors.black,
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: 18,
    fontWeight: 'bold',
    lineHeight: 24,
  },
});

export default styles;
