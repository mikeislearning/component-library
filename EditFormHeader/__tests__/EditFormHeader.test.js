import React from 'react';
import { fireEvent, render } from 'react-native-testing-library';

import EditFormHeader from '../EditFormHeader';

describe('<EditFormHeader />', () => {
  it('renders correctly', () => {
    const header = render(
      <EditFormHeader onClose={() => {}} onSave={() => {}} title="Title Text" />
    );
    expect(header).toMatchSnapshot();
  });

  it('renders without a title if it is not passed', () => {
    const header = render(<EditFormHeader onClose={() => {}} onSave={() => {}} />);
    expect(header).toMatchSnapshot();
  });

  it('can handle pressing the Save button successfully', () => {
    const mockedFn = jest.fn();
    const { getByText } = render(
      <EditFormHeader onClose={() => {}} onSave={mockedFn} title="Title Text" isValidForm={true} />
    );

    fireEvent.press(getByText('Save'));

    expect(mockedFn).toHaveBeenCalledTimes(1);
  });

  it('can handle pressing the Close button successfully', () => {
    const mockedFn = jest.fn();
    const { getByType } = render(
      <EditFormHeader onClose={mockedFn} onSave={() => {}} title="Title Text" />
    );

    fireEvent.press(getByType('Image'));

    expect(mockedFn).toHaveBeenCalledTimes(1);
  });
});
