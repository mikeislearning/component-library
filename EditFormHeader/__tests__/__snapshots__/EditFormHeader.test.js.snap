// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditFormHeader /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#FBFBFB",
      "borderBottomColor": "#ECECEC",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 57,
      "justifyContent": "center",
      "paddingBottom": 12,
    }
  }
>
  <View
    accessibilityLabel="Close"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 12,
        "left": 0,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
        "position": "absolute",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/images/close.png",
        }
      }
      style={
        Object {
          "height": 17,
          "width": 17,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "MessinaSans-Regular",
        "fontSize": 18,
        "fontWeight": "bold",
        "lineHeight": 24,
      }
    }
  >
    Title Text
  </Text>
  <View
    style={
      Object {
        "bottom": 12,
        "paddingRight": 20,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#4145C8",
            "fontFamily": "MessinaSans-SemiBold",
            "fontSize": 18,
            "lineHeight": 24,
          },
          Object {
            "color": "#726E67",
          },
        ]
      }
    >
      Save
    </Text>
  </View>
</View>
`;

exports[`<EditFormHeader /> renders without a title if it is not passed 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#FBFBFB",
      "borderBottomColor": "#ECECEC",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 57,
      "justifyContent": "center",
      "paddingBottom": 12,
    }
  }
>
  <View
    accessibilityLabel="Close"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 12,
        "left": 0,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
        "position": "absolute",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/images/close.png",
        }
      }
      style={
        Object {
          "height": 17,
          "width": 17,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "bottom": 12,
        "paddingRight": 20,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#4145C8",
            "fontFamily": "MessinaSans-SemiBold",
            "fontSize": 18,
            "lineHeight": 24,
          },
          Object {
            "color": "#726E67",
          },
        ]
      }
    >
      Save
    </Text>
  </View>
</View>
`;
